api/v1beta1/verticadb_conversion.go

```diff
@@ -0,0 +1,398 @@
+		UpgradeStatus:   src.UpgradeStatus,
+	}
+	for i := range src.Subclusters {
+		dst.Subclusters[i] = convertFromSubclusterStatus(src.Subclusters[i])
+	}
+	for i := range src.Conditions {
+		dst.Conditions[i] = convertFromStatusCondition(src.Conditions[i])
+	}
+	return dst
+}
+
+// convertToSubcluster will take a v1beta1 Subcluster and convert it to a v1 version
+func convertToSubcluster(src *Subcluster) v1.Subcluster {
+	return v1.Subcluster{
+		Name:                src.Name,
+		Size:                src.Size,
+		IsPrimary:           src.IsPrimary,
+		IsTransient:         src.IsTransient,
+		ImageOverride:       src.ImageOverride,
+		NodeSelector:        src.NodeSelector,
+		Affinity:            v1.Affinity(src.Affinity),
+		PriorityClassName:   src.PriorityClassName,
+		Tolerations:         src.Tolerations,
+		Resources:           src.Resources,
+		ServiceType:         src.ServiceType,
+		ServiceName:         src.ServiceName,
+		NodePort:            src.NodePort,
+		VerticaHTTPNodePort: src.VerticaHTTPNodePort,
+		ExternalIPs:         src.ExternalIPs,
+		LoadBalancerIP:      src.LoadBalancerIP,
+		ServiceAnnotations:  src.ServiceAnnotations,
+	}
+}
+
+// convertFromSubcluster will take a v1 Subcluster and convert it to a v1beta1 version
+func convertFromSubcluster(src *v1.Subcluster) Subcluster {
+	return Subcluster{
+		Name:                src.Name,
+		Size:                src.Size,
+		IsPrimary:           src.IsPrimary,
+		IsTransient:         src.IsTransient,
+		ImageOverride:       src.ImageOverride,
+		NodeSelector:        src.NodeSelector,
+		Affinity:            Affinity(src.Affinity),
+		PriorityClassName:   src.PriorityClassName,
+		Tolerations:         src.Tolerations,
+		Resources:           src.Resources,
+		ServiceType:         src.ServiceType,
+		ServiceName:         src.ServiceName,
+		NodePort:            src.NodePort,
+		VerticaHTTPNodePort: src.VerticaHTTPNodePort,
+		ExternalIPs:         src.ExternalIPs,
+		LoadBalancerIP:      src.LoadBalancerIP,
+		ServiceAnnotations:  src.ServiceAnnotations,
+	}
+}
+
+// convertToCommunal will convert to a v1 VerticaDBCondition from a v1beta1 version
+func convertToCommunal(src *CommunalStorage) v1.CommunalStorage {
+	return v1.CommunalStorage{
+		Path:                   src.Path,
+		IncludeUIDInPath:       src.IncludeUIDInPath,
+		Endpoint:               src.Endpoint,
+		CredentialSecret:       src.CredentialSecret,
+		CaFile:                 src.CaFile,
+		Region:                 src.Region,
+		KerberosServiceName:    src.KerberosServiceName,
+		KerberosRealm:          src.KerberosRealm,
+		S3ServerSideEncryption: v1.ServerSideEncryptionType(src.S3ServerSideEncryption),
+		S3SseCustomerKeySecret: src.S3SseCustomerKeySecret,
+		AdditionalConfig:       src.AdditionalConfig,
+	}
+}
+
+// convertFromCommunal will convert from a v1 CommunalStorage to a v1beta1 version
+func convertFromCommunal(src *v1.CommunalStorage, hadoopConfig string) CommunalStorage {
```

Comment:
How are you going to handle this when there will be more changes?

Reply:
Yeah, good point. This file will need to be refactored as more changes are added.